examples/face_recognition_11/face_recognition_11/evaluator.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import numpy as np
+from face_recognition_11.workflow import FMRConfiguration
+from face_recognition_11.workflow import GroundTruth
+from face_recognition_11.workflow import Inference
+from face_recognition_11.workflow import TestCase
+from face_recognition_11.workflow import TestCaseMetrics
+from face_recognition_11.workflow import TestSample
+from face_recognition_11.workflow import TestSampleMetrics
+
+from kolena.workflow import AxisConfig
+from kolena.workflow import Curve
+from kolena.workflow import CurvePlot
+from kolena.workflow import Evaluator
+from kolena.workflow import Plot
+
+
+class FaceRecognition11Evaluator(Evaluator):
+    @staticmethod
+    def compute_threshold(inferences: List[Inference], fmr: float) -> float:
+        """
+        The threhsold is calculated as Threshold = Quantile_v(1 - FMR) where the quantiles are of the observed
+        IMPOSTER scores, v.
+
+        For more details on how threshold is used see https://pages.nist.gov/frvt/reports/11/frvt_11_report.pdf
+        """
+        similarity_scores = np.array([inf.similarity for inf in inferences])
+        threshold = np.quantile(similarity_scores, 1 - fmr)
+        return threshold
+
+    @staticmethod
+    def compute_test_sample_metrics_single(
+        ground_truth: GroundTruth,
+        inference: Inference,
+        threshold: float,
+    ) -> TestSampleMetrics:
+        is_match, is_false_match, is_false_non_match = False, False, False
+
+        if inference.similarity is None:
+            return TestSampleMetrics(
+                is_match=is_match,
+                is_false_match=is_false_match,
+                is_false_non_match=is_false_non_match,
+                failure_to_enroll=True,
+            )
+
+        if inference.similarity > threshold:  # match
+            if ground_truth.is_same:
+                is_match = True
+            else:
+                is_false_match = True
+        else:  # no match
+            if ground_truth.is_same:
+                is_false_non_match = True
+
+        return TestSampleMetrics(
+            is_match=is_match,
+            is_false_match=is_false_match,
+            is_false_non_match=is_false_non_match,
+            failure_to_enroll=False,
+        )
+
+    def compute_test_sample_metrics(
```

Comment:
consider using [evaluator function](https://docs.kolena.io/reference/workflow/evaluator/#kolena.workflow.evaluator_function.BasicEvaluatorFunction) instead of evaluator class as it computes the test sample metrics once for each configuration.

Reply:
Another benefit of using evaluator function is that we can be sure that we are computing threshold with the complete set of test samples. 

With the class implementation, `compute_test_sample_metrics` is called for each and every test case.  So, what is happening here is that a threshold is computed for each test case.

Reply:
Yeah, I tried with the evaluator function...but the problem is that I need to compute a new threshold across every new test sample since I'm using the similarity scores that come only from a test sample.

Reply:
> I need to compute a new threshold across every new test sample

Isn't threshold computed against the complete test case? Shouldn't there be only one threshold per configuration for a test suite?

Reply:
Sorry. I meant that it's computed across each test case not test sample since I'm thinking the complete test case and every other test case would have their own threshold.

Reply:
Shouldn't the threshold be computed with the complete test case only and the same threshold should be used on evaluating all test cases?